src/tests/utils/setup.ts

```diff
@@ -0,0 +1,130 @@
+import { INestApplication, Module, Inject } from '@nestjs/common'
+import { Test, TestingModuleBuilder } from '@nestjs/testing'
+import { getEntityManagerToken, getConnectionToken, TypeOrmModule } from '@nestjs/typeorm'
+import { merge as _merge } from 'lodash'
+import { AdminCoreModuleFactory } from '../../adminCore.module'
+import DefaultAdminSite from '../../adminSite'
+import { TestTypeOrmModule } from './testTypeOrmModule'
+import { TestAuthModule } from './testAuth.module'
+import { EntityType } from '../../types'
+import { injectionTokens } from '../../tokens'
+import { User } from '../../../exampleApp/src/user/user.entity'
+import { Group } from '../../../exampleApp/src/user/group.entity'
+import { Agency } from '../../../exampleApp/src/user/agency.entity'
+
+export interface TestApplication extends INestApplication {
+  startTest: () => Promise<void>
+  stopTest: () => Promise<void>
+  testStarted: boolean
+  doClose: INestApplication['close']
+}
+
+function createRegistrationModule(registrables: EntityType[], adminModules: any[]) {
+  @Module({ imports: [...adminModules] })
+  class AdminRegistrationModule {
+    constructor(@Inject(injectionTokens.ADMIN_SITE) private readonly adminSite: DefaultAdminSite) {
+      registrables.forEach(registrable => {
+        adminSite.register('test', registrable)
```

Comment:
do we not have any tests that check that adminSites with more than one section display as expected? maybe `{section: string, entity: EntityType}[]` instead?